recipes/docker_registry.rb

```diff
@@ -148,3 +148,18 @@
   action :delete
   only_if { File.exist? "#{Chef::Config['file_cache_path']}/#{base_filename}" }
 end
+
+# We add docker in kagent in this recipe as the hops::docker recipe runs during the install phase and it might run
+# before kagent::install
+service_name='docker'
+if node['kagent']['enabled'] == "true"
+  kagent_config service_name do
+    service "docker"
+  end
+end
+
+if conda_helpers.is_upgrade
+  kagent_config "#{service_name}" do
```

Comment:
Wouldn't `  kagent_config service_name do` as above work?

Reply:
All usages of  `conda_helpers.is_upgrade` in our codebase do it like this, I assume it's a bad code example that has propagated through time. In any case, I fixed it here.